### PR TITLE
Feature/board

### PR DESCRIPTION
**Pull request by ghyl**

### CODE REVIEW

**Comment by LeonWubben** [comment]
Prima

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java [hunk 1]
```diff
@@ -36,7 +46,59 @@ public Board copy(Board b) {
...
+    }
+
+    public ArrayList<Piece> removeAllConnectedLines() {
+        removeAllLRows();
+        removeAllTRows();
+        removeAllFiveRows();
+        removeAllFourRows();
+        removeAllThreeRows();
+        ArrayList<Piece> result = (ArrayList<Piece>) removedPieces.clone();
+        System.out.println(result);
```
Comment (LeonWubben) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java]: System.out

File: src/test/java/nl/tudelft/ti2206/semgr8/model/BoardTest.java [hunk 2]
```diff
@@ -135,53 +141,387 @@ public void testSwap(){
...
+    }
+
+    @Test
+    public void testCopy(){
+        Board b = Board.getRandomBoard();
+        Board res = board.copy(b);
+        assertTrue(b.equals(res));
+    }
+
+//    @Test(expected=RuntimeException.class)
```
Comment (albert-smit) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/BoardTest.java]: why are these tests commented out?

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java [hunk 3]
```diff
@@ -2,24 +2,34 @@
...
 import java.util.Arrays;
 import java.util.Optional;
-import java.util.Random;
 import java.util.Scanner;
 
 /**
  * Represents the Bejeweled board with pieces and executes all operations on the board (such as
  * swapping pieces, removing pieces, etc.).
  */
+@SuppressWarnings("checkstyle:magicnumber")
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Board.java]: please add a comment explaining why this is needed

File: src/main/java/nl/tudelft/ti2206/semgr8/model/Piece.java [hunk 4]
```diff
@@ -17,12 +18,13 @@ public int getColor() {
...
     }
 
-
-
     public void setColor(int color) {
         if (color >= 0 && color < AMOUNT_OF_COLORS) {
             this.color = color;
         }
+        else{
+            throw new IndexOutOfBoundsException();
```
Comment (albert-smit) [inline src/main/java/nl/tudelft/ti2206/semgr8/model/Piece.java]: please add an error message to this exception

**Comment by albert-smit** [comment]
looks good but, Board.read still relies on default encoding, and EmptyPiece only has an equals method, and no hashCode.

File: src/test/java/nl/tudelft/ti2206/semgr8/model/BoardTest.java [hunk 5]
```diff
@@ -14,11 +14,26 @@
...
     @Before
     public void setUp() {
-        board = new Board(path);
+        board = new Board(pathToGrid1);
+    }
+
+    @Test
+    public void testRemoveAllLines() {
+        board = new Board(pathToGridWithConnectedLines);
+        System.out.println(board);
```
Comment (albert-smit) [inline src/test/java/nl/tudelft/ti2206/semgr8/model/BoardTest.java]: another debug print

**Pull request merged**